csrc/kernel.cpp

```diff
@@ -315,6 +315,11 @@ void Kernel::finalize(std::vector<Expr*> top_level_exprs) {
   summary_.sync_map = GpuLower::current()->syncMap();
   summary_.parallel_dimension_map_ =
       GpuLower::current()->parallelDimensionMap();
+  parameters_ = GpuLower::current()->allKnownVals();
+  parameters_.insert(parameters_.end(), outputs().begin(), outputs().end());
```

Comment:
Yes, exactly, this can be used for supporting scalar outputs. I will not touch the `ValType == TensorView` in this PR because it is better handled in the PR that enables the scalar output support.